### PR TITLE
Gradle Build Implementation

### PR DESCRIPTION
**Pull request by matthewchivers**
Initial work towards building the Galasa framework using Gradle.

### CODE REVIEW

**Comment by matthewchivers** [comment]
PR marked as draft due to needing to investigate implication of global sub-project dependencies.

> Potential implications of these dependencies were noticed just before opening the PR. Early reviews welcome, but more commits are likely.

**Comment by matthewchivers** [comment]
All jar-producing bundles now have a Gradle build mechanism in place that produces an equivalent jar to that of then maven output (in some places, an exact match hasn't been possible, but work has been undertaken to reduce differences as much as reasonably possible).

Manifest files (`META-INF/MANIFEST.MF`) appear to be the main difference between the generated artifacts. For example, the Gradle plugin populates the header `Private-Package` with any packages that aren't explicitly exported. The same header doesn't appear in the artifact produced by Maven. 
Equally, the Gradle output manifest lacks the header: `Embedded-Artifacts` which appears to be maven-plugin specific and isn't a part of the OSGi specification.

**Comment by Michael-Baylis** [comment]
Ok,  will have a look.   Might take a while.

File: galasa-parent/galasa-boot/build.gradle [hunk 1]
```diff
@@ -0,0 +1,58 @@
+plugins {
+    id 'biz.aQute.bnd.builder' version '5.1.2'
```
Comment (Michael-Baylis) [inline galasa-parent/galasa-boot/build.gradle:2]: Not required,   boot is not an OSGi project as it starts up OSGi

**Review by Michael-Baylis — CHANGES_REQUESTED** [review]
two extremely minor changes.   VERY good job

File: galasa-parent/galasa-testharness/build.gradle [hunk 2]
```diff
@@ -0,0 +1,8 @@
+description = 'Galasa Boot'
```
[the remark this answers is not in the document]
Comment (matthewchivers) [reply galasa-parent/galasa-testharness/build.gradle:1]: Corrected.

File: galasa-parent/galasa-boot/build.gradle (hunk 1 above)
Comment (matthewchivers) [reply galasa-parent/galasa-boot/build.gradle]: Corrected.

**Review by Michael-Baylis — APPROVED** [review]

**Pull request merged**